src/server/compositor/multi_threaded_compositor.h

```diff
@@ -74,8 +74,8 @@ class MultiThreadedCompositor : public Compositor
     void destroy_compositing_threads();
 
     std::shared_ptr<graphics::Display> const display;
-    std::shared_ptr<Scene> const scene;
     std::shared_ptr<DisplayBufferCompositorFactory> const display_buffer_compositor_factory;
+    std::shared_ptr<Scene> const scene;
```

Comment:
This is the only active ingredient, right? Moving `scene` to be destroyed before `display_buffer_compositor_factory` which incidentally has shared ownership of the platforms?

Parameter order in the constructor isn't relevant, so we don't need all the other changes?

Reply:
Agreed the parameter order isn't an essential part of the fix: only the data member order and the initialiser order are required. But an inconsistency between the parameter order and the initialiser order gives the reader something unnecessary to think about: it is simpler with the parameter order change.

Reply:
I agree, the _change_ is more complex, but the result is not.